### ISSUE TITLE
Resume Copying works very bad

### ISSUE THREAD

**Issue by berserker465** [ericyd/gdrive-copy#2]
Hello

I tried to copy a folder with almost 20.000 files on Google Drive, however the copy process stopped....so I resumed it using the Resume option and it correctly resumed....for like 10 minutes, then it stopped again.

Long story short:the resume feature works yes, but it keeps stopping over and over, for copying all the said files I had to resume the copy process roughly over 30 times.

While I think its surely better than nothing, a way to automatic resume an interrupted process would really makes this script perfect.

**Comment by ericyd** [comment]
The resume function typically works better when several hours have elapsed since the time the app stopped copying. My working hypothesis is that when the app stops like that, it is due to a daily quota that has been exceeded. I believe the extra time allows the quota to reset.
I know this is not ideal, and I am open to suggestions on how to improve this process.

---- On Mon, 31 Oct 2016 10:24:36 -0700 notifications@github.com wrote ----

Hello

I tried to copy a folder with almost 20.000 files on Google Drive, however the copy process stopped....so I resumed it using the Resume option and it correctly resumed....for like 10 minutes, then it stopped again.

Long story short:the resume feature works yes, but it keeps stopping over and over, for copying all the said files I had to resume the copy process roughly over 30 times.

While I think its surely better than nothing, a way to automatic resume an interrupted process would really makes this script perfect.

**Comment by berserker465** [comment]
May I suggest to increase the pause time based on the number of files being copied?

For example,after coping 5.000+ files the script could pause for 60 minutes instead of the 1-2 minutes, then continue for another 5.000 files with the usual 1-2 minutes pause, then again a 60 minutes pause,and so on.

**Comment by ericyd** [comment]
That is a really great suggestion! I'm not sure when I'll have time to implement it, but I think this would likely solve a lot of these cases.

**Comment by tmalaher** [comment]
I had to copy about 15000 files/dirs. The first 4000 or so worked fine then it paused for a day before I realized what was up. After that it paused several more times before it finally completed.

So some sort of exponential back-off might be good. Are you able to detect if the resume works/fails, or is it a "throw it over the wall and hope" situation?

**Comment by berserker465** [comment]
I'm no expert in compiling scripts, but I'm guessing this could be fixed with a string that checks if the number of copied files hasn't been changed for over 10 minutes and if the operation is still incomplete, then it would restart it automatically.
But I think that the main problem is a script can't restart itself if its freezes....I'm just making a guess here.

**Comment by ericyd** [comment]
This is essentially a duplicate of #31 and that one has a more descriptive title so I'm going to close this issue for now.
